Src/Private/Resolve-AzDoModuleUri.ps1

```diff
@@ -0,0 +1,55 @@
+function Resolve-AzDoModuleUri {
+ <#
+    .SYNOPSIS
+        Returns the direct download Uri for a PowerShell module hosted
+        on the Azure DevOps Artifacts.
+#>
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param (
+        ## PowerShell DSC resource module name
+        [Parameter(Mandatory, ValueFromPipeline, ValueFromPipelineByPropertyName)] [ValidateNotNullOrEmpty()]
+        [System.String] $Name,
+
+        ## The minimum version of the DSC module required
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [ValidateNotNullOrEmpty()]
+        [System.Version] $MinimumVersion,
+
+        ## The exact version of the DSC module required
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [ValidateNotNullOrEmpty()]
+        [System.Version] $RequiredVersion,
+
+        ## Direct download Uri
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [ValidateNotNullOrEmpty()]
+        [System.String] $Uri,
+
+        ## Catch all, for splatting $PSBoundParameters
+        [Parameter(ValueFromRemainingArguments)]
+        $RemainingArguments
+    )
+    process {
+
+        if ($PSBoundParameters.ContainsKey('Uri')) {
+
+            $psRepositoryUri = $Uri;
+        }
+        else {
+
+            $psRepositoryUri = (Get-ConfigurationData -Configuration Host).RepositoryUri;
+        }
+
+        if ($PSBoundParameters.ContainsKey('RequiredVersion')) {
+           ## Download the specific version
+           return ('{0}?id={1}&version={2}' -f $psRepositoryUri, $Name.ToLower(), "$($RequiredVersion.Major).$($RequiredVersion.Minor).$($RequiredVersion.Build)")
```

Comment:
Could the module version also be `1.2` or `1.2.3.4`?

Reply:
I ran into another problem here. If you call the AzDo feed with `?id=ModuleName?version=1.2.3` then everything is OK. However, if you do not specify a version then `?id=ModuleName` returns an XML file and __not__ the latest version.

This config works:
```
@{
    AllNodes = @(
        @{
            NodeName = '2019
        }
    )
    NonNodeData = @{
        Lability = @{
            Module = @(
                @{ Name = 'VirtualEngine.AppDeployment'; Provider = 'AzDo'; RequiredVersion = '2.2.2' }
            )
        }
    }
}
```
Whereas, this config does not work:
```
@{
    AllNodes = @(
        @{
            NodeName = '2019
        }
    )
    NonNodeData = @{
        Lability = @{
            Module = @(
                @{ Name = 'VirtualEngine.AppDeployment'; Provider = 'AzDo'; Latest = $true }
            )
        }
    }
}
```
We'll need to refer to the docs to work out how to retrieve the latest version as I tried `?id=ModuleName?version=` and `?id=ModuleName?version=*` to no avail.